src/sentry/static/sentry/app/components/group/sentryAppExternalIssueForm.tsx

```diff
@@ -39,6 +56,27 @@ export class SentryAppExternalIssueForm extends React.Component<Props> {
     event: SentryTypes.Event,
     onSubmitSuccess: PropTypes.func,
   };
+  state: State = {};
+
+  componentDidMount() {
+    this.resetStateFromProps();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.action !== this.props.action) {
```

Comment:
@billyvg Action is either `create` or `link` issue. When the value of `action` changes, the form re-mounts and calls `reset` on the model. That being said, it might be smart to just call the model reset explicitly here as well in the case in the future for whatever reason the form does not re-mount when `action` changes.

Reply:
Are you sure it remounts? If it did, you wouldn't need this lifecycle method

Reply:
@billyvg The child component `form` remounts but not this component. That's why I had to make that change in the form model for the `reset` method when it would set `api` to `null`.

Reply:
You can force this component to remount by giving it `key={action}` where you render this component, I think that might be cleaner than trying to sync state, that might solve the reset issue as well.

React will consider the components different because the `key` is different, and instead of updating, will unmount and mount the "new" component.

Reply:
🔑

Reply:
Lol this actually didn't help because the component doesn't load the options when it mounts, even with `autoLoad=true`

Reply:
@billyvg So I actually can't reset the form state when we switch action because it clears out these hidden meta fields we have. This whole form is a mess...